src/uint/mul.rs

```diff
@@ -9,6 +9,11 @@ use crate::{
 use core::ops::{Mul, MulAssign};
 use subtle::CtOption;
 
+#[cfg(feature = "alloc")]
+mod karatsuba;
```

Comment:
Since this is gated under `alloc` and not shared with `Uint`, maybe just put it under `uint/boxed/karatsuba.rs`?

Reply:
well my goal was to have this work with `Uint` before merging, just trying to get the first thing working

Reply:
Aah ok. Right now sharing code involves macros like `impl_schoolbook_multiplication!` since without `const_mut_refs` there's no way to construct anything other than a const generic-sized array in a `const fn`